app/controllers/SocialAuthController.scala

```diff
@@ -34,9 +36,20 @@ class SocialAuthController @Inject() (
             user <- userService.save(profile)
             _ <- authInfoRepository.save(profile.loginInfo, authInfo)
             authenticator <- silhouette.env.authenticatorService.create(profile.loginInfo)
-            value <- silhouette.env.authenticatorService.init(authenticator)
-            result <- silhouette.env.authenticatorService.embed(value, Redirect(userState.state("redirect")))
+            token <- silhouette.env.authenticatorService.init(authenticator)
+            result <- silhouette.env.authenticatorService.embed(token, Redirect(userState.state("redirect")))
           } yield {
+
+            val url = routes.ActivateAccountController.activate(token).absoluteURL()
+            if (!user.activated) {
+              mailerClient.send(Email(
```

Comment:
Is it worth encapsulating instantiation of this email in a separate function since it appears to basically same as in `SignUpController`?

Reply:
how do you feel about punting unless we either have to instantiate another Email like this or find some other duplicative functionality?